client/components/app.jsx

```diff
@@ -9,26 +9,77 @@ export default class App extends React.Component {
     this.state = {
       view: {
         name: 'catalog',
-        params: {}
+        params: {},
+        cart: []
       }
     };
     this.setView = this.setView.bind(this);
+    this.getCartItems = this.getCartItems.bind(this);
+    this.addToCart = this.addToCart.bind(this);
+
+  }
+
+  componentDidMount() {
+    this.getCartItems();
   }
 
   setView(name, params) {
     this.setState({
       view: {
         name: name,
-        params: params
+        params: params,
+        cart: this.state.view.cart
       }
     });
+
+  }
+
+  addToCart(product) {
+
+    fetch('api/cart', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(product)
+    })
+      .then(response => response.json())
+      .then(data => {
+        this.state.view.cart.push(data);
```

Comment:
Be careful not to mutate state directly. If you need to add to an array, there are a few ways to to it:
- slice & push
- concat
- spread